File: src/main/java/com/moovie/backend/domain/repository/ProfileRepository.java
package com.moovie.backend.domain.repository;

import com.moovie.backend.domain.model.PublicProfile;
import com.moovie.backend.domain.model.UserProfile;

public interface ProfileRepository {

    UserProfile getUserProfile();

    void updateUserProfile(UserProfile profile);

    PublicProfile getPublicProfile(String userId);
}

File: src/main/kotlin/org/mobyle/domain/repository/ProfileRepository.kt
package org.mobyle.domain.repository

import org.mobyle.domain.model.PublicProfile
import org.mobyle.domain.model.UserProfile

interface ProfileRepository {
    suspend fun getUserProfile(): UserProfile
    suspend fun updateUserProfile(profile: UserProfile)
    suspend fun getPublicProfile(userId: String): PublicProfile
}
